### PR TITLE
Application - Make Experience step fully functional

### LINKED ISSUE #2860: Applicant - Make My Timeline step of new Application fully functional
# Description
Making the step functional means tying it into the redux store and APIs so that it get can GET and UPDATE application data, as well as navigate to other pages (SAVE AND EXIT or CONTINUE).

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #2860.

Incorporates #4395, so ideally merge that first.

### Notes:
- The handleSubmit function is quite complicated, because the API call is different depending on whether a new Experience must be created or an existing one must be updated, and because each ExperienceSkill object must be created/deleted with an individual API request. This could be simplified, but would require more work on the backend.

### CODE REVIEW

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Looks good so far 👍 Just a couple issues: 
- The experience skills aren't getting added to the db. 
- I made a fix for a mistake I made on the Experience component when displaying contants on a couple of the accordions (education and award).

File: resources/assets/js/components/Checkbox.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ const Checkbox: React.FunctionComponent<CheckboxProps> = ({
     <>
       {checkboxGroup ? (
         <div data-c-grid-item={grid}>
-          <label htmlFor={id}>
+          <label>
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/components/Checkbox.tsx:56]: Yeah, and the htmlFor caused more issues when there were non-unique ids on the page. Obviously we should avoid non-unique ids anyway, but I don't think there's a downside to removing this.

**Comment by tristan-orourke** [comment]
Regarding how the experience constants were showing up in experience accordions, good catch that things weren't working. 
Chris and I were planning on a different solution, where we add the text of the constants directly in the Experience api responses, and then we can simply use them on the frontend. 
So I'm going to complete those API response and then revert 062671b44cbd448bdb98ca982c08879320c464ff .

File: resources/assets/js/components/Application/Experience/ExperiencePage.tsx [hunk 2]
```diff
@@ -105,16 +358,17 @@ export const ExperiencePage: React.FC<ExperiencePageProps> = ({
         experienceSkills={experienceSkills}
         criteria={criteria}
         skills={skills}
-        jobId={job.id}
-        jobClassificationId={job.classification_id ?? 1}
-        recipientTypes={recipientTypes}
-        recognitionTypes={recognitionTypes}
+        jobId={job?.id ?? 1}
+        jobClassificationId={job?.classification_id ?? 1}
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/components/Application/Experience/ExperiencePage.tsx]: I've added a loading state for when job isn't loading yet, and I've allowed the code to gracefully handle classification_id being null, though that should never be the case for a published job poster.

**Review by yonikid15 — COMMENTED** [review]
Looks good, good job 👍  
Just one thing. The modal forms need to be reset after submission `actions.resetForm()`.
Also, we need to fix the focus issue on the CheckboxField component, then merge that branch first.

**Comment by Khristophor** [comment]
@tristan-orourke `JobWorkEnv.test.tsx` snapshot and `Job Poster Builder - New Job` Testcafe test need updating, and then this should be good to go

**Review by yonikid15 — APPROVED** [review]

**Pull request merged**